test/core/transport/chaotic_good/client_transport_test.cc

```diff
@@ -0,0 +1,390 @@
+  auto message = arena_->MakePooled<Message>(std::move(buffer), 0);
+  ClientMetadataHandle md;
+  auto args = CallArgs{
+      std::move(md), ClientInitialMetadataOutstandingToken::Empty(), nullptr,
+      nullptr,       &pipe_client_to_server_messages_.receiver,      nullptr};
+  StrictMock<MockFunction<void(absl::Status)>> on_done;
+  EXPECT_CALL(on_done, Call(absl::OkStatus()));
+  EXPECT_CALL(control_endpoint_, Write)
+      .WillOnce(
+          WithArgs<0>([this](absl::AnyInvocable<void(absl::Status)> on_write) {
+            on_write(this->kDummyErrorStatus);
+            return false;
+          }));
+  EXPECT_CALL(data_endpoint_, Write)
+      .WillOnce(
+          WithArgs<0>([this](absl::AnyInvocable<void(absl::Status)> on_write) {
+            on_write(this->kDummyErrorStatus);
+            return false;
+          }));
+  auto activity = MakeActivity(
+      Seq(
+          // Concurrently: send message into the pipe, and receive from the
+          // pipe.
+          Join(Seq(pipe_client_to_server_messages_.sender.Push(
+                       std::move(message)),
+                   [this] {
+                     this->pipe_client_to_server_messages_.sender.Close();
+                     return absl::OkStatus();
+                   }),
+               client_transport_.AddStream(std::move(args))),
+          // Once complete, verify successful sending and the received value.
+          [](const std::tuple<absl::Status, absl::Status>& ret) {
+            // TODO(ladynana): change these expectations to errors after the
+            // writer activity closes transport for EE failures.
+            EXPECT_TRUE(std::get<0>(ret).ok());
+            EXPECT_TRUE(std::get<1>(ret).ok());
+            return absl::OkStatus();
+          }),
+      InlineWakeupScheduler(),
+      [&on_done](absl::Status status) { on_done.Call(std::move(status)); });
+  // Wait until ClientTransport's internal activities to finish.
+  event_engine_->TickUntilIdle();
+  event_engine_->UnsetGlobalHooks();
+}
+
+TEST_F(ClientTransportTest, AddOneStreamMultipleMessages) {
+  SliceBuffer buffer;
+  buffer.Append(Slice::FromCopiedString("test add stream."));
+  auto message = arena_->MakePooled<Message>(std::move(buffer), 0);
+  ClientMetadataHandle md;
+  auto args = CallArgs{
+      std::move(md), ClientInitialMetadataOutstandingToken::Empty(), nullptr,
+      nullptr,       &pipe_client_to_server_messages_.receiver,      nullptr};
+  StrictMock<MockFunction<void(absl::Status)>> on_done;
+  EXPECT_CALL(on_done, Call(absl::OkStatus()));
+  EXPECT_CALL(control_endpoint_, Write).Times(3).WillRepeatedly(Return(true));
+  EXPECT_CALL(data_endpoint_, Write).Times(3).WillRepeatedly(Return(true));
+  auto activity = MakeActivity(
+      Seq(
+          // Concurrently: send messages into the pipe, and receive from the
+          // pipe.
+          Join(Seq(pipe_client_to_server_messages_.sender.Push(
+                       std::move(message)),
```

Comment:
Could this cause a use-after free error ? You seem to be std::move the same message 3 times. Maybe you meant to allocate a different message each time

Reply:
I didn't see the use-after free error, but I think you are right about moving the same message 3 times will cause problem. The latter 2 moves will point to nullptr after the first move. I have changed it to allocate different messages. 

There's an issue in test build which cause the checks are failed after sync to the master branch. I am debugging the root cause.